ipfs/ipfs.go

```diff
@@ -0,0 +1,316 @@
+	ifacepath "github.com/ipfs/interface-go-ipfs-core/path"
+	ma "github.com/multiformats/go-multiaddr"
+
+	"github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/core/coreapi"
+	"github.com/ipfs/go-ipfs/core/node/libp2p"
+	"github.com/ipfs/go-ipfs/plugin/loader" // This package is needed so that all the preloaded plugins are loaded automatically
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+	"github.com/libp2p/go-libp2p-core/peer"
+	status "github.com/sonr-io/core/errors"
+)
+
+var FlagExp = flag.Bool("experimental", false, "enable experimental features")
+
+const tempDir string = "/tmp/bytes.txt"
+const ipfsStorageDir string = "/ipfs-storage"
+
+func setupPlugins(externalPluginsPath string) error {
+	// Load any external plugins if available on externalPluginsPath
+	plugins, err := loader.NewPluginLoader(filepath.Join(externalPluginsPath, "plugins"))
+	if err != nil {
+		return fmt.Errorf("error loading plugins: %s", err)
+	}
+
+	// Load preloaded and external plugins
+	if err := plugins.Initialize(); err != nil {
+		return fmt.Errorf("error initializing plugins: %s", err)
+	}
+
+	if err := plugins.Inject(); err != nil {
+		return fmt.Errorf("error initializing plugins: %s", err)
+	}
+
+	return nil
+}
+
+func createTempRepo() (string, error) {
+	repoPath, err := ioutil.TempDir("", "ipfs-shell")
+	if err != nil {
+		return "", fmt.Errorf("failed to get temp dir: %s", err)
+	}
+
+	// Create a config with default options and a 2048 bit key
+	cfg, err := config.Init(ioutil.Discard, 2048)
+	if err != nil {
+		return "", err
+	}
+
+	// When creating the repository, you can define custom settings on the repository, such as enabling experimental
+	// features (See experimental-features.md) or customizing the gateway endpoint.
+	// To do such things, you should modify the variable `cfg`. For example:
+	if *FlagExp {
+		// https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-filestore
+		cfg.Experimental.FilestoreEnabled = true
+		// https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-urlstore
+		cfg.Experimental.UrlstoreEnabled = true
+		// https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-p2p
+		cfg.Experimental.Libp2pStreamMounting = true
+		// https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#p2p-http-proxy
+		cfg.Experimental.P2pHttpProxy = true
+		// https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#strategic-providing
+		cfg.Experimental.StrategicProviding = true
+	}
+
+	// Create the repo with the config
+	err = fsrepo.Init(repoPath, cfg)
+	if err != nil {
+		return "", fmt.Errorf("failed to init ephemeral node: %s", err)
+	}
+
+	return repoPath, nil
+}
+
+/// ------ Spawning the node
+
+// Creates an IPFS node and returns its coreAPI
+func createNode(ctx context.Context, repoPath string) (iface.CoreAPI, error) {
+	// Open the repo
+	repo, err := fsrepo.Open(repoPath)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
might want to add some debug logging with the open operation fails.

Reply:
looking into fsrepo.open it seems like they have lots of internal checks that woudl get returned in err. It woudl be idiomatic to just log this err variable by the caller. I don't think extra debug logging would be particularly helpful if already covered in the .Open function (such as path not existing, permission mismatch, etc)

Reply:
Resolving.